File: src/main/java/fr/tikione/c2e/service/web/CPCAuthService.java
package fr.tikione.c2e.service.web;

import fr.tikione.c2e.model.web.Auth;

/**
 * Tools to authenticate on CanardPC website.
 */
public interface CPCAuthService {
    
    /**
     * Authenticate on CanardPC website.
     * @param username username.
     * @param password password.
     * @return authentication data, to be reused by future requests.
     */
    Auth authenticate(String username, String password);
}

File: app/src/main/java/fr/tikione/c2e/service/web/CPCAuthService.kt
package fr.tikione.c2e.service.web

import fr.tikione.c2e.model.web.Auth

/**
 * Tools to authenticate on CanardPC website.
 */
interface CPCAuthService {

    /**
     * Authenticate on CanardPC website.
     * @param username username.
     * @param password password.
     * @return authentication data, to be reused by future requests.
     */
    fun authenticate(username: String, password: String): Auth
}
